config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        </module>
+        <module name="InnerAssignment" />
+        <module name="MagicNumber" />
+        <module name="MissingCtor" />
+        <module name="MissingSwitchDefault" />
+        <module name="ModifiedControlVariable" />
+        <module name="MultipleStringLiterals" />
+        <module name="MultipleVariableDeclarations" />
+        <module name="NestedForDepth" />
+        <module name="NestedIfDepth" />
+        <module name="NestedTryDepth" />
+        <module name="NoClone" />
+        <module name="NoFinalizer" />
+        <module name="OneStatementPerLine" />
+        <module name="OverloadMethodsDeclarationOrder" />
+        <module name="PackageDeclaration" />
+        <module name="ParameterAssignment" />
+        <module name="RequireThis" />
+        <module name="ReturnCount" />
+        <module name="SimplifyBooleanExpression" />
+        <module name="SimplifyBooleanReturn" />
+        <module name="StringLiteralEquality" />
+        <module name="SuperClone" />
+        <module name="SuperFinalize" />
+        <module name="UnnecessaryParentheses" />
+        <module name="VariableDeclarationUsageDistance" />
 
-    <module name="TreeWalker">
+        <!-- Checks for imports                              -->
+        <!-- See http://checkstyle.sourceforge.net/config_import.html -->
+        <module name="AvoidStarImport" />
+        <!-- We suppress this warning because the Java creators themselves disagree on this one, for example in the newest JUnit 5 -->
+        <module name="AvoidStaticImport">
+            <property name="severity" value="ignore" />
+        </module>
+        <module name="IllegalImport" />
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportControl"/> -->
+        <!-- Suppressed because we don't have so many imports that this is necessary -->
+        <!-- <module name="ImportOrder"/> -->
+        <module name="RedundantImport" />
+        <module name="UnusedImports" />
 
         <!-- Checks for Javadoc comments.                     -->
         <!-- See http://checkstyle.sourceforge.net/config_javadoc.html -->
         <module name="JavadocMethod">
             <property name="scope" value="package" />
-            <property name="allowMissingParamTags" value="true" />
+            <property name="allowMissingPropertyJavadoc" value="true" />
             <property name="allowMissingThrowsTags" value="true" />
-            <property name="allowMissingReturnTag" value="true" />
             <property name="allowThrowsTagsForSubclasses" value="true" />
             <property name="allowUndeclaredRTE" value="true" />
-            <property name="allowMissingPropertyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
         </module>
         <module name="JavadocType">
             <property name="scope" value="package" />
-            <property name="severity" value="ignore" />
```

Comment:
We don't ignore JavadocType because I think that most classes also need some Javadoc. Literally all large professional Java programs document their classes so that everybody understand what these classes are for and I think we should do the same for that reason